pkg/controller/daemon/update.go

```diff
@@ -71,6 +71,13 @@ func (dsc *DaemonSetsController) rollingUpdate(ctx context.Context, ds *apps.Dae
 		var allowedReplacementPods []string
 		var candidatePodsToDelete []string
 		for nodeName, pods := range nodeToDaemonPods {
+
+			// skip this loop, if node is not Ready
```

Comment:
If the new pod caused the node to go not ready, isn’t “pausing” the desired behavior?

Example:

1. Run network agent with daemonset
2. Every node is ready
3. Update daemonset with new version of network agent
4. First node it goes to causes a kernel panic
5. Admin expects the daemonset not to rollout to any new nodes (otherwise all nodes go down)

This seems like it is intentional to not skip those nodes, and some users may depend on this